src/components/Feeds/FeedPage/FeedComments/ReplyCard.jsx

```diff
@@ -14,44 +14,65 @@ import { dateFormatter } from "src/components/Common/dateFormatter";
-import Options from "src/components/Common/ModalWindowOptions";
+import Options from "src/components/Common/ModalOptions";
+import { deleteFeed, updateFeed } from "src/features/feeds/feedsSlice";
+import { useDispatch } from "react-redux";
+import ModifyFeed from "src/components/Feeds/PostForm/ModifyFeed";
 
 const ReplyCard = ({ reply, user, comments, likes, bookmarks, views, displayAt, updateFeedView }) => {
+    const [editMode, setEditMode] = useState(false);
     const avatar = cdnContentImagesUrl("/user/" + (reply?.avatar || "avatarDummy.png"));
-    // const dispatch = useDispatch();
-    const handleCommentDelete = () => {};
-
+    const dispatch = useDispatch();
+    const handleCommentDelete = () => {
+        dispatch(deleteFeed(reply._id));
+    };
+    const handleEditFeed = () => {
+        setEditMode(true);
+    };
+    const handleSaveEditedFeed = (data) => {
+        dispatch(updateFeed({ id: reply?._id, feedData: data }));
+    };
     return (
         <FeedPostContainer $displayAt={displayAt}>
             <LeftSection>
                 <PostHeaderImg src={avatar} alt={reply?.username + `avatar`} />
             </LeftSection>
-            <RightSection>
-                {/* <RouteLink to={`/feeds/${reply?._id}`}> */}
-                <PostHeader>
-                    <LeftSection>
-                        <PostHeaderUsername>{reply?.username}</PostHeaderUsername>
-                        {reply?.verified && <IconVerified />}
-                    </LeftSection>
-                    <RightHeaderSection>
-                        <PostTimestamp>{dateFormatter({ date: new Date(reply?.createdAt) })}</PostTimestamp>
-                        {user && user._id === reply.user && <Options onDelete={handleCommentDelete} />}
-                    </RightHeaderSection>
-                </PostHeader>
-                <PostContent>{reply?.reply}</PostContent>
-                {/* </RouteLink> */}
+            {editMode ? (
+                <ModifyFeed showPostTags={true} editFeed={reply} onModifyFeed={handleSaveEditedFeed} />
+            ) : (
+                <RightSection>
+                    {/* <RouteLink to={`/feeds/${reply?._id}`}> */}
+                    <PostHeader>
+                        <LeftSection>
+                            <PostHeaderUsername>{reply?.username}</PostHeaderUsername>
+                            {reply?.verified && <IconVerified />}
+                        </LeftSection>
+                        <RightHeaderSection>
+                            <PostTimestamp>{dateFormatter({ date: new Date(reply?.createdAt) })}</PostTimestamp>
+                            {user && user._id === reply.user && (
+                                <Options
+                                    onDelete={handleCommentDelete}
+                                    onEdit={handleEditFeed}
+                                    modalContainerId={"reply-card-options-container-" + reply._id}
+                                />
```

Comment:
Nice, it will work but :
I think we can do it a bit more elegant with the "cooler" syntax `reply-card-options-container- ${reply._id}`

Reply:
what do you think ?